relentless/optimize/method.py

```diff
@@ -145,15 +153,33 @@ class LineSearch:
 
         \mathbf{d} = \mathbf{x}_{end} - \mathbf{x}_{start}
 
-        \hat{\mathbf{d}} = \frac{\mathbf{d}}{\lVert\mathbf{d}\rVert}
+        \mathbf{\hat{d}} = \frac{\mathbf{d}}{\lVert\mathbf{d}\rVert}
+
+    :math:`\mathbf{X}=\left[X_1,\ldots,X_n\right]` is defined as the scaling
+    parameters for the gradient such that :math:`y_i=\frac{x_i}{X_i}`. By
+    default, the variables are left unscaled (:math:`X_i=1`).
+
```

Comment:
Ok so I think lines 145-160 can be simplified to something like "Given an ObjectiveFunction f(x) and a search interval x_start and x_end, LineSearch seeks an optimal step 0 < alpha < 1 such that f(x_start + alpha d) is minimized, were the search direction d and hat d is...."

Reply:
I like this, I think this explains what is going on nicely.